module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -264,4 +264,23 @@ public function createEntity(): Resource
         $class = $this->getEntityClass(Resource::class);
         return new $class;
     }
+
+    /**
+     * Get a set of records that do not have metadata stored in the resource
+     * table.
+     *
+     * @return array|null
+     */
+    public function findMissingMetadata()
+    {
+        $dql = "SELECT r "
+            . "FROM " . $this->getEntityClass(Resource::class) . " r "
+            . "WHERE r.title = :title OR r.author IS NULL OR r.year IS NULL";
```

Comment:
Since we're searching for a fixed empty value here, I don't think we need to use a placeholder -- we can just hard-code `r.title = ''` into the DQL and remove the `setParameters` call and related variables. Placeholders should only be needed when we're sanitizing variable input. (I checked the DQL reference manual and saw several examples of hard-coded strings in DQL, so I'm pretty confident it's normal practice).